Documentation/botr/shared-generics.md

```diff
@@ -0,0 +1,185 @@
+Shared Generics Design
+===
+
+Author: Fadi Hanna - 2019
+
+# Introduction
+
+Shared generics is a runtime+JIT feature aimed at reducing the amount of code the runtime generates for generic methods of various instantiations (supports methods on generic types and generic methods). The idea is that for certain instantiations, the generated code will almost be identical with the exception of a few instructions, so in order to reduce the memory footprint, and the amount of time we spend jitting these generic methods, the runtime will generate a single special canonical version of the code, which can be used by all compatible instantiations of the method.
```

Comment:
It may be worth referencing the original MSR paper https://www.microsoft.com/en-us/research/publication/design-and-implementation-of-generics-for-the-net-common-language-runtime/ that explains the design choices and reasons for the chosen design.

Reply:
Ah, thank you! I didn't know we had a document on this. I'll make the update.